site/content/deployment/enmasse.md

```diff
@@ -0,0 +1,56 @@
++++
+title = "OpenShift"
+weight = 475
++++
+
+## Build Hono
+
+~~~sh
+~/hono$ mvn clean install -Pbuild-docker-image
+~~~
+
+## Deploy EnMasse
+
+Dowload the release, unpack and run
+
+~~~sh
+./deploy-openshift.sh -n hono
+~~~
+
+## Deploy Hono
+
+~~~sh
+~hono/example/target/deploy/openshift$ chmod +x *
+~hono/example/target/deploy/openshift$ ./enmasse_deploy.sh
+~~~
+
+## Extract Certificate
+
+~~~sh
+~hono/example$ oc extract secret/external-certs-messaging --to=target/config/hono-demo-certs-jar/ -n hono
+~~~
+
+## Start Consumer
+
+~~~sh
+~hono/example$ mvn spring-boot:run -Drun.arguments=--hono.client.host=messaging-hono.$(minishift ip).nip.io,--hono.client.port=443,--hono.client.trustStorePath=target/config/hono-demo-certs-jar/server-cert.pem,--hono.client.username=consumer@HONO,--hono.client.password=verysecret
+~~~
+
+## Register Device
+
+~~~sh
+~hono/example$ curl -X POST -i -H 'Content-Type: application/json' -d '{"device-id": "4711"}' http://hono-service-device-registry-http-hono.$(minishift ip).nip.io/registration/DEFAULT_TENANT
```

Comment:
I think the node port is missing, isn't it?

Reply:
The current OpenShift deployment guide access to the Device Registry using the minishift-ip + node port, @dejanb is using the route exposed by OpenShift via HTTPS and port 443 ... never tried, does it work ?

Reply:
The routes work like this, there's no need for port

Reply:
Ah right, OpenShift supports even clear HTTP without TLS, it doesn't support TCP without TLS ! Right ;)